File: UL/System/Char.cs
namespace System
{
    public struct Char
    {
        public static readonly char MaxValue = 0xFFFF;
        public static readonly char MinValue = 0;


        public extern static Char Parse(string value);

        public extern override string ToString();

        public static bool TryParse(string value, out Char v)
        {
            try
            {
                v = Parse(value);
                return true;
            }
            catch (Exception e)
            {
                v = 0;
                return false;
            }
        }

        public virtual Boolean op_Equals(Char b);
        public virtual Boolean op_Small(Char b);
        public virtual Char op_Assign(Char b);
        public virtual Char op_PlusPlus(Char b);
    }
}

File: ul/CppApp/Char.cpp
#include "stdafx.h"
#include "Char.h"
#include "Object.h"
#include "String.h"
#include "Boolean.h"
#include "Exception.h"
System::Char System::Char::MaxValue=0xFFFF;
System::Char System::Char::MinValue=0;
System::Boolean System::Char::TryParse(Ref<System::String>  value,System::Char  v)
{
	try
	{
		v = Parse(value);
		return true;
	}
	catch(System::Exception e)
	{
		v = 0;
		return false;
	}
}
